### ISSUE TITLE
Reading list progress inconsistency

### ISSUE THREAD

**Issue by tb94** [Kareadita/Kavita#2966]
### What happened?

I have a reading list I'm working through, but every time try to continue where I left off it puts me several issues back on the last page. I've tried marking them as read, even marking them as unread and swiping through to the end, but I can't seem to get my progress tracked correctly.
last night I read a few comics from the reading list (after finding my place again), and when I press continue on my list today It takes me back almost to the beginning of the list (but shows me the last page of the comic as if I read every page but the last) and swiping through I see the last page of each comic from there until I get to where I left off.
I read a few more today, then went to check on a new series I just got, then went back to my reading list and got put back to the same (wrong) comic on the last page.

images show the reading list progress does not match the (completed) progress shown in the series page.

![series_progress](https://github.com/Kareadita/Kavita/assets/36962804/2f3acc6a-f0a2-4d1c-b595-92f4855167d0)
![reading_list_progress](https://github.com/Kareadita/Kavita/assets/36962804/0c447aa9-7a49-4ad7-a2f1-0e7f90d2b033)

### What did you expect?

Reading list progress is tracked correctly and is reflective of progress in the series page. Continuing a reading list after leaving the page should continue from the previous reading session.

### Kavita Version Number - If you don not see your version number listed, please update Kavita and see if your issue still persists.

0.8.1 - Stable

### What operating system is Kavita being hosted from?

Docker (Dockerhub Container)

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

None

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

Chrome

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

**Comment by tb94** [comment]
follow up - I deleted the reading list and marked the `Journey into Mystery` series as unread, then imported my CBL again, and I lost almost all progress on issues in the reading list. it looks like only one comic in the list is still marked as read.

**Comment by tb94** [comment]
I was able to reproduce this today by re-reading a comic that appears earlier in the reading list from where my current progress is. for example: I'm currently at position 10 in the reading list. I went back to the comic at position 4 to reference an event that was called back in comic 10. upon continuing the reading list I'm placed on the last page of comic 4 instead of my previous position of comic 10.

**Comment by majora2007** [comment]
When you went back to position 4, did you read to completion or left some reading progress?

**Comment by majora2007** [comment]
Okay I can't reproduce. What I did:

1. Add Chapters 1-10 to a new reading list. There is no reading progress existing
2. Start reading from the reading list button. Read all of Chapter 1. 
3. Progressed into Chapter 2 via continuous reading (did not leave reader). 
4. Backed out and hit Continue from reading list. This loaded Chapter 2. 

I checked in the series and the progress matched what is happening with the reading list (as it should, as progress is only at the chapter level and the reading list just has a custom order).
